runtime/onert/backend/train/ops/PoolLayer.cc

```diff
@@ -25,7 +31,104 @@ namespace train
 namespace ops
 {
 
-PoolLayer::PoolLayer() : cpu::ops::PoolLayer()
+namespace
+{
+
+cpu::ops::PoolType convert_to_infer(const train::ops::PoolType &pool_type)
+{
+  switch (pool_type)
+  {
+    case train::ops::PoolType::kMax:
+      return cpu::ops::PoolType::kMax;
+    default:
+      throw std::runtime_error("PoolLayer: Unsupported pool type yet");
+  }
+}
+
+class MaxPool2D final : public TrainingKernelRegistry
```

Comment:
This class is a wrapper of  "max pool" related things. 

It provides
	* forward and backward interfaces for maxpooling (call proper kernel) 
	* holds additional memory used to record the max-arguments' index.